public/app/features/plugins/admin/components/PluginDetailsRightPanel.tsx

```diff
@@ -13,6 +20,37 @@ type Props = {
 
 export function PluginDetailsRightPanel(props: Props): React.ReactElement | null {
   const { info, plugin } = props;
+  const [defaultBranch, setDefaultBranch] = useState<string | null>(null);
+
+  const trailURLSlash = (url: string | null) => (url?.endsWith('/') ? url?.slice(0, -1) : url);
+
+  const repositoryLink = trailURLSlash(
+    plugin.url ||
+      (plugin.signatureType === 'community' && plugin.details?.links
+        ? getPluginRepositoryUrl(plugin.details.links)
+        : null)
+  );
+
+  useEffect(() => {
+    if (repositoryLink) {
+      fetchDefaultBranchFromRepo(repositoryLink)
+        .then(setDefaultBranch)
+        .catch(() => setDefaultBranch(null));
+    }
+  }, [repositoryLink]);
+
+  const licenseLink = plugin.details?.links
```

Comment:
in this logic, having `plugin.details?.links` doesn't guarantee we'll have a license, documentation or custom links at all we are potentially creating 404 links here because we are falling back to the repository link but there's no guarantee those exist,